### PR TITLE
Add 'How It Works section in README'

### PR DESCRIPTION
**Pull request by fborello-lambda**
- Add Diagram
- Explain Abstraction CosmosSDK offers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,3 +35,14 @@ lambchaind query tx <txhash>
 
 The blockchain in development can be configured with `config.yml`.
 
+# How It Works
+
+A blockchain can be created using the igniteCLI, which collects various files from CosmosSDK, making it easier to deploy a blockchain to production. CosmosSDK manages communication between the Application Layer and the Consensus Layer via the ABCI. The Consensus Layer, in turn, is responsible for CometBFT, which uses Tendermint as the consensus client.
+
+In the Application Layer, messages are defined to be included in transactions. The application then forwards the message's information to a Module, which is essentially a Go function triggered when the message is included in a transaction.
+
+Abstracting these concepts, we have the following diagram: a person sends a transaction to the blockchain, the blockchain processes it, and redirects the information to the appropriate module based on the message's information. The module we aim to implement is a Snarks validator, so the transaction needs to include the proof, the validator key, and the public witness.
```
Comment (JulianGCalderon) [inline README.md:44]: I think this paragraphs belong to another section, where we can explain in depth the lifecycle of the transaction.

File: imgs/Diagram_Cosmos.svg
Comment (JulianGCalderon) [inline imgs/Diagram_Cosmos.svg:1]: The sequence is confusing. The transaction should enter the node through the networking module and work it's way up to the application and then to our custom module.

CometBFT contains both the consensus and the networking module.

**Review by JulianGCalderon — CHANGES_REQUESTED** [review]

**Review by JulianGCalderon — APPROVED** [review]
LGTM

**Review by tomyrd — APPROVED** [review]
LGTM

**Pull request merged**